pages/performances/index.js

```diff
@@ -18,62 +22,302 @@ import styles from '@styles/pages/Performances.module.scss'; // Page styles
 // Temp constants
 import data, { columns } from '../../data/mockParticipants';
 
+const PAGE_SIZE = 3; // Rows per page
+
+// Get the active filters (list of column accessors) from an object of filter dropdown values
+const getActiveFilters = options => {
+  return Object.keys(options).filter(option => options[option].selected);
+};
+
+// Remove key from object (returns new object)
+const removeKeyFromObject = (object, key) => {
+  // eslint-disable-next-line no-unused-vars
+  const { [key]: _, ...rest } = object;
+  return rest;
+};
+
 // Page: Performances
 export default function Performances() {
   const [modalOpen, setModalOpen] = useState(false);
   const [showFilters, setShowFilters] = useState(false);
-  // const entryContent = <Table></Table>;
+  const [query, setQuery] = useState('');
+  const [schoolOptions, setSchoolOptions] = useState({});
+  const [academicLevelOptions, setAcademicLevelOptions] = useState({});
+  const [performanceLevelOptions, setPerformanceLevelOptions] = useState({});
+  const [danceStyleOptions, setDanceStyleOptions] = useState({});
+  const [danceSizeOptions, setDanceSizeOptions] = useState({});
+  const [filters, setFilters] = useState([]);
+  const [pageNumber, setPageNumber] = useState(0);
+  const [pageCount, setPageCount] = useState(0);
+
+  // Get initial filter options
+  useEffect(() => {
+    const getSettings = async () => {
```

Comment:
Kinda confused here - is this supposed to be in the `page/performances/index.js` file?

Reply:
Yes, sorry about the confusing naming. Setting (especially school, dance style and dance level) are used to filter the performances, so they need to be retrieved to populate the filter dropdowns.

Reply:
Ahh so to get the different options for each category, right? Sounds good then